source/nodejs/adaptivecards-designer/src/card-designer.ts

```diff
@@ -449,9 +457,21 @@ export class CardDesigner extends Designer.DesignContext {
     }
 
     private targetVersionChanged() {
+        // Question: is there a specific reason that we get the payload from the designerSurface first?
         let cardPayload = this.designerSurface.getCardPayloadAsObject();
 
-        if (typeof cardPayload === "object") {
+        if (this._isPayloadCardArray) {
+            cardPayload = JSON.parse(this.getCurrentCardEditorPayload());
+        }
+
+        if (Array.isArray(cardPayload)) {
+
+            for (let card of cardPayload) {
```

Comment:
what will happen if there are multiple cards after `$when` evaluation?

Reply:
If there are multiple cards after expanding the template, the templating library will log an error and return undefined.

At this point in the code, we are updating the contents of the JSON editor so that each card contains the correct version; so `$when` evaluation hasn't occurred yet.